code/modules/detectivework/evidence.dm

```diff
@@ -79,6 +79,15 @@
 		icon_state = "evidenceobj"
 	return
 
+obj/item/weapon/evidencebag/attackby(obj/item/weapon/W as obj, mob/living/user as mob)
+	if(istype(W, /obj/item/weapon/pen))
+		var/new_label = input("What should the new label be?","")
```

Comment:
var/new_label = sanitize(trim(input("What should the new label be", "") as null|text))